operators/endpointmetrics/pkg/openshift/openshift.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+// Licensed under the Apache License 2.0
+
+package openshift
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	ocinfrav1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ClusterRoleBindingName = "metrics-collector-view"
+	CaConfigmapName        = "metrics-collector-serving-certs-ca-bundle"
+	OwnerLabelKey          = "owner"
+	OwnerLabelValue        = "observabilityaddon"
+)
+
+var (
+	log = ctrl.Log.WithName("controllers").WithName("ObservabilityAddon")
+)
```

Comment:
Can we avoid this global? It has a pretty common name that is often used for imports (`log`) and could be passed via parameter to the functions (even via context).

Reply:
As I was refactoring this, I was hopping to remove them completely. But it actually requires changing some functions signatures to reflect back some status info. Doing a second pass